pkgs/development/libraries/livesplit-core/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, dynamicLib ? true
+, generateBindings ? true
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "livesplit-core";
+  version = "unstable-2021-08-23";
+
+  src = fetchFromGitHub {
+    owner = "livesplit";
+    repo = pname;
+    rev = "6fb02347f0ec4db8fb4bd8e284f8a8bc71e15686";
+    sha256 = "0279hmlcdpbd5g6x02vp8gq8zj8j37kzw6d5k892rp5fr4z5xkga";
+  };
+
+  # Upstream does not provide a lockfile for cargo dependency versions, so we use our own.
```

Comment:
Please add some more explanation here. You should mention that
- this is a library, so it's not customary to include a `Cargo.lock` in the upstream repository.
- we need locked dependencies for packaging.
- we don't usually package rust libraries (I think? I'm not an expert here) but we need to package this one since a non-rust package (redtimer) depends on it.

Basically everything that was discussed [here](https://github.com/NixOS/nixpkgs/pull/135479#discussion_r698872792). You could also link that discussion.